include/fiction/algorithms/physical_design/graph_oriented_layout_design.hpp

```diff
@@ -0,0 +1,1523 @@
+//
+// Created by Simon Hofmann on 30.01.24.
+//
+#ifndef FICTION_GRAPH_ORIENTED_LAYOUT_DESIGN_HPP
+#define FICTION_GRAPH_ORIENTED_LAYOUT_DESIGN_HPP
+
+#include "fiction/algorithms/network_transformation/fanout_substitution.hpp"
+#include "fiction/algorithms/path_finding/a_star.hpp"
+#include "fiction/algorithms/path_finding/cost.hpp"
+#include "fiction/algorithms/path_finding/distance.hpp"
+#include "fiction/algorithms/physical_design/post_layout_optimization.hpp"
+#include "fiction/io/print_layout.hpp"
+#include "fiction/layouts/bounding_box.hpp"
+#include "fiction/layouts/clocking_scheme.hpp"
+#include "fiction/layouts/obstruction_layout.hpp"
+#include "fiction/networks/technology_network.hpp"
+#include "fiction/traits.hpp"
+#include "fiction/types.hpp"
+#include "fiction/utils/name_utils.hpp"
+#include "fiction/utils/network_utils.hpp"
+#include "fiction/utils/placement_utils.hpp"
+#include "fiction/utils/routing_utils.hpp"
+
+#include <fmt/format.h>
+#include <mockturtle/traits.hpp>
+#include <mockturtle/utils/node_map.hpp>
+#include <mockturtle/utils/stopwatch.hpp>
+#include <mockturtle/views/fanout_view.hpp>
+#include <mockturtle/views/immutable_view.hpp>
+#include <mockturtle/views/names_view.hpp>
+
+#include <array>
+#include <cassert>
+#include <chrono>
+#include <cstddef>
+#include <cstdint>
+#include <functional>
+#include <iostream>
+#include <optional>
+#include <queue>
+#include <string>
+#include <tuple>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+namespace fiction
+{
+
+/**
+ * Parameters for the graph-oriented layout design algorithm.
+ */
+struct graph_oriented_layout_design_params
+{
+    /**
+     * High effort mode.
+     */
+    bool high_effort = false;
+    /**
+     * Return first found layout.
+     */
+    bool return_first = false;
+    /**
+     * Timeout limit (in ms).
+     */
+    std::optional<uint64_t> timeout = std::nullopt;
```

Comment:
We usually don't use `std::optional` for timeout even though that is a nice idea. Instead, we initialize the value with `std::numeric_limits<uint64_t>::max()`.